pkg/virt-controller/watch/vmi_test.go

```diff
@@ -1859,6 +1859,75 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 			controller.Execute()
 		})
 
+		DescribeTable("should update pod labels when VMI dynamic label set changes", func(vmiLabels map[string]string, podLabels map[string]string, expectedPatch string) {
+			vmi := NewPendingVirtualMachine("testvmi")
+			vmi.Status.Phase = virtv1.Running
+
+			pod := NewPodForVirtualMachine(vmi, k8sv1.PodRunning)
+
+			vmi.Labels = vmiLabels
+			for key, val := range podLabels {
+				pod.Labels[key] = val
+			}
+
+			addVirtualMachine(vmi)
+			addActivePods(vmi, pod.UID, "")
+			podFeeder.Add(pod)
+
+			vmiInterface.EXPECT().Patch(context.Background(), vmi.Name, types.JSONPatchType, gomock.Any(), &metav1.PatchOptions{}).Return(vmi, nil)
+
+			patch := ""
+			kubeClient.Fake.PrependReactor("patch", "pods", func(action testing.Action) (handled bool, obj k8sruntime.Object, err error) {
+				p, _ := action.(testing.PatchAction)
+				patch = string(p.GetPatch())
+				return true, nil, nil
+			})
+
+			key := kvcontroller.VirtualMachineInstanceKey(vmi)
+			err := controller.execute(key)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(patch).To(Equal(expectedPatch))
+		},
+			Entry("when VMI and pod labels differ",
+				map[string]string{
+					virtv1.NodeNameLabel: "node2",
+				},
+				map[string]string{
+					virtv1.NodeNameLabel: "node1",
+				},
+				`[{ "op": "test", "path": "/metadata/labels", "value": {"kubevirt.io":"virt-launcher","kubevirt.io/created-by":"1234","kubevirt.io/nodeName":"node1"} }, { "op": "replace", "path": "/metadata/labels", "value": {"kubevirt.io":"virt-launcher","kubevirt.io/created-by":"1234","kubevirt.io/nodeName":"node2"} }]`,
+			),
+			Entry("when VMI and pod label are the same",
+				map[string]string{
+					virtv1.NodeNameLabel: "node1",
+				},
+				map[string]string{
+					virtv1.NodeNameLabel: "node1",
+				},
+				"",
+			),
```

Comment:
Is it worth it to use a test data struct to put names to test fields? 
```golang
Entry("when VMI and pod label are the same", testData{
				vmiLabels: map[string]string{
					virtv1.NodeNameLabel: "node1",
				},
				podLabels: map[string]string{
					virtv1.NodeNameLabel: "node1",
				},
				expectedPatch: "",
                           },
			),
```

Reply:
Probably not in this case, simply due to the establish precedence set by the rest of the test table entries in this unit test file. I'd favor consistency in structure over diverging from the pattern used throughout the rest of the file.

Reply:
oh, I misinterpreted your comment, yeah this would make sense. I'll updated it.

Reply:
fixed